sunpy/map/sources/yohkoh.py

```diff
@@ -57,23 +57,22 @@ def detector(self):
     @property
     def dsun(self):
         """
-        For Yohkoh Maps, DSUN_OBS is not always defined. In this case the
-        SOLAR_R keyword is used to calculate dsun.
-        """
+        For Yohkoh Maps, DSUN_OBS is not always defined.
 
+        In this case the SOLAR_R keyword is used to calculate dsun.
+        """
         # 2012/12/19 - the SXT headers do not have a value of the distance from
-        # the spacecraft to the center of the Sun.  The FITS keyword 'DSUN_OBS'
-        # appears to refer to the observed diameter of the Sun.  Until such
+        # the spacecraft to the center of the Sun. The FITS keyword 'DSUN_OBS'
+        # appears to refer to the observed diameter of the Sun. Until such
         # time as that is calculated and properly included in the file, we will
         # use simple trigonometry to calculate the distance of the center of
-        # the Sun from the spacecraft.  Note that the small angle approximation
+        # the Sun from the spacecraft. Note that the small angle approximation
         # is used, and the solar radius stored in SXT FITS files is in arcseconds.
-
         if 'solar_r' in self.meta:
             dsun = constants.radius / (np.deg2rad(self.meta['solar_r'] / 3600.0))
         else:
             dsun = constants.au
-        return self.meta.get('dsun_obs', dsun)
+        return self.meta.get('dsun_obs', dsun) * u.m
```

Comment:
I had to do this.

Reply:
Er, neither is correct.  `self.meta.get('dsun_obs')` is a bare value (in meters), while `dsun` is already a `Quantity`.

Reply:
I will fix this.

Reply:
What do you think about the comment above it? 

If that is true, we don't in fact want to return the meta keyword value at all?